data/file_level_aggregation.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import pandas as pd
+from typing import List, Tuple
+from datetime import timedelta
+from data.constants import *
+from data.utils import *
+
+
+class HistoricalData:
+    def __init__(self) -> None:
+        """Init HistoricalData."""
+        self.file_name = None
+        self.repo_name = None
+        self.data = defaultdict(list)
+
+    def __repr__(self) -> str:
+        """Return the str representation of FileData.
+
+        Returns:
+            A str representation of data dict.
+        """
+        return str(self.data)
+
+
+class DataAggregator:
+    def __init__(self, file_level_data: pd.DataFrame) -> None:
+        """Inits DataAggregator with the CSV file name.
+
+        Args:
+            file: The pandas data frame of file level data.
+        """
+        self._pr_related_columns = PULL_REQUEST_RELATED_COLUMNS
+        self._file_related_columns = FILE_RELATED_COLUMNS
+        self._file_level_data = file_level_data
+        self._file_level_data = self._file_level_data[
+            self._file_level_data['file name'].notna()]
```

Comment:
Why you are reassigning self._file_level_data

Reply:
Some corresponding files of the review comments no longer exist in the record (removed and not merged to the upstream) so their file names are nan and need to be removed. But yes, I can refactor this code to be `self._file_level_data = file_level_data[file_level_data['file name'].notna()]` so I dont need to reassign the self._file_level_data